### PR TITLE
Clean BuildFiles in  RecoHI and RecoLocalMuon

### PR DESCRIPTION
**Pull request by guitargeek**
#### PR description:

Another quick round of automatic BuildFile.xml cleaning to improve the modularity of CMSSW.

#### PR validation:

CMSSW compiles. It was checked that all newly added dependencies are actually required by the package with `git grep`.

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by kpedro88** [comment]
+upgrade

File: RecoHI/HiEvtPlaneAlgos/BuildFile.xml [hunk 1]
```diff
@@ -1,23 +1,17 @@
...
 <use   name="DataFormats/VertexReco"/>
 <use   name="DataFormats/Common"/>
 <use   name="DataFormats/HeavyIonEvent"/>
 <use   name="DataFormats/CaloTowers"/>
-<use   name="DataFormats/Provenance"/>
 <use   name="CondCore/DBOutputService"/>
 <use   name="CondFormats/DataRecord"/>
-<use   name="RecoHI/HiCentralityAlgos"/>
-<use   name="clhep"/>
+<use   name="SimDataFormats/TrackingAnalysis"/>
```
Comment (slava77) [inline RecoHI/HiEvtPlaneAlgos/BuildFile.xml]: this looks like a fake dependency.
May I ask you to cleanup the includes in RecoHI/HiEvtPlaneAlgos/src/HiEvtPlaneFlatProducer.cc which apparently does not need anything from Sim\*.

Comment (guitargeek) [reply RecoHI/HiEvtPlaneAlgos/BuildFile.xml]: You are right, these Sim dependencies in Reco are not really good. I'll remove it.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/28972 58e74e3 
- code changes are in line with the PR description and the follow up review: cleanup of dependencies and some unused files or includes
- jenkins tests pass and comparisons with the baseline show no differences

**Comment by kpedro88** [comment]
+upgrade

**Comment by silviodonato** [comment]
+1

**Pull request merged**